datasets/cfq/cfq.py

```diff
@@ -131,22 +130,36 @@ def _scrub_json(self, content):
         # For the 4GB dataset file it requires more than 40GB of RAM and takes 3min.
         # There are more efficient libraries but in order to avoid additional
         # dependencies we use a simple (perhaps somewhat brittle) regexp to reduce
-        # the content to only what is needed. This takes 1min to execute but
-        # afterwards loading requires only 500MB or RAM and is done in 2s.
-        regex = re.compile(r'("%s":\s*"[^"]*").*?("%s":\s*"[^"]*")' % (_QUESTION_FIELD, _QUERY_FIELD), re.DOTALL)
-        return "[" + ",".join(["{" + m.group(1) + "," + m.group(2) + "}" for m in regex.finditer(content)]) + "]"
-
-    def _generate_examples(self, base_directory, splits_file, split_id):
+        # the content to only what is needed.
+        question_regex = re.compile(r'("%s":\s*"[^"]*")' % _QUESTION_FIELD)
+        query_regex = re.compile(r'("%s":\s*"[^"]*")' % _QUERY_FIELD)
+        question_match = None
+        result = []
+        for line in content:
+            line = line.decode("utf-8")
+            if not question_match:
+                question_match = question_regex.match(line)
+            else:
+                query_match = query_regex.match(line)
+                if query_match:
+                    result.append((question_match.group(1), query_match.group(1)))
+                    question_match = None
+        return "[" + ",".join(["{" + match[0] + "," + match[1] + "}" for match in result]) + "]"
```

Comment:
maybe this can be done iteratively to not have all the dataset in RAM
```suggestion
                    yield json.loads("{" + question_match.group(1) + "," + query_match.group(1) + "}")
                    question_match = None
```

Reply:
Ah good catch. I think it's fine to merge it as is then for now

Though ideally the dataset needs to be restructured to allow a practical streaming and lower RAM usage.